perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/impl/ConsolidatorManagerImpl.java

```diff
@@ -472,36 +486,21 @@ private UserExtSource createExtSourceAndUserExtSource(User user, String actor, S
 	 * @return application object / null if not exists
 	 */
 	private Application getLatestApplication(PerunSession sess, Vo vo, Group group, Application.AppType type) {
-		try {
-
-			if (sess.getPerunPrincipal().getUser() != null) {
-
-				if (group != null) {
-
-					return jdbc.queryForObject(RegistrarManagerImpl.APP_SELECT + " where a.id=(select max(id) from application where vo_id=? and group_id=? and apptype=? and user_id=? )", RegistrarManagerImpl.APP_MAPPER, vo.getId(), group.getId(), String.valueOf(type), sess.getPerunPrincipal().getUserId());
-
-				} else {
-
-					return jdbc.queryForObject(RegistrarManagerImpl.APP_SELECT + " where a.id=(select max(id) from application where vo_id=? and apptype=? and user_id=? )", RegistrarManagerImpl.APP_MAPPER, vo.getId(), String.valueOf(type), sess.getPerunPrincipal().getUserId());
-
-				}
-
-			} else {
-
-				if (group != null) {
-
-					return jdbc.queryForObject(RegistrarManagerImpl.APP_SELECT + " where a.id=(select max(id) from application where vo_id=? and group_id=? and apptype=? and created_by=? and extsourcename=? )", RegistrarManagerImpl.APP_MAPPER, vo.getId(), group.getId(), String.valueOf(type), sess.getPerunPrincipal().getActor(), sess.getPerunPrincipal().getExtSourceName());
-
-				} else {
-
-					return jdbc.queryForObject(RegistrarManagerImpl.APP_SELECT + " where a.id=(select max(id) from application where vo_id=? and apptype=? and created_by=? and extsourcename=? )", RegistrarManagerImpl.APP_MAPPER, vo.getId(), String.valueOf(type), sess.getPerunPrincipal().getActor(), sess.getPerunPrincipal().getExtSourceName());
-
-				}
+		List<Application> allApplications = new ArrayList<>();
+		if (group != null) {
+			allApplications.addAll(jdbc.query(RegistrarManagerImpl.APP_SELECT + " where a.vo_id=? and a.group_id=? and a.apptype=?", RegistrarManagerImpl.APP_MAPPER, vo.getId(), group.getId(), String.valueOf(type)));
+		} else {
+			allApplications.addAll(jdbc.query(RegistrarManagerImpl.APP_SELECT + " where a.vo_id=? and a.apptype=?", RegistrarManagerImpl.APP_MAPPER, vo.getId(), String.valueOf(type)));
```

Comment:
Not your change and I'm not sure it matters, but second select retrieve also all group applications in that vo, since ``group_id is null`` is not part of the SQL.
Result is later used to get similar users. So depending on the application content it might be more or less successful in finding similar users. I guess using only VO applications, one might be more successful. I'm not sure its relevant though, since I don't know callers (outside) context.